### PR TITLE
Added API Contract test case for servers endpoint and refactored existing ones.

### PR DESCRIPTION
**Pull request by gokulakrishnansvm**
- Added Api contract test case for server endpoint.
- Refactored code for all existing Api contract test cases, Used Json schema validation for validating Keys and data types of values in the test case.
- Replaced response_template.json file with json schema formats.
- Refactored all fixtures(post_data) in conftest, Added common method to verfiy response template and added common methods for get and post requests.
- Moved request_template.json , response_template.json and to_data.json to folder named data.
<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Ops Test

## What is the best way to verify this PR?

Step 1 : Follow steps from README file(traffic_ops/testing/api_contract) to run api-contract-tests.

Step 2 : To execute servers API Contract Testcase , navigate to traffic_ops/testing/api_contract/v4
pytest --to-user Username --to-password Password --to-url URL --config Config Path --request-template Request template --response-template Response template test_servers.py

Note: Config, Request template and response template paths are optional. 

## PR submission checklist
- [x] This PR has tests 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: traffic_ops/testing/api_contract/v4/conftest.py [hunk 1]
```diff
@@ -845,39 +770,78 @@ def phys_locations_post_data(to_session: TOSession, request_template_data: list[
 		if not isinstance(name, str):
 			raise TypeError(f"name must be str, not '{type(name)}'")
 		phys_locations["name"] = name[:4] + randstr
-		shortName = phys_locations["shortName"]
+		shortname = phys_locations["shortName"]
```
Comment (TerekhovaKate) [inline traffic_ops/testing/api_contract/v4/conftest.py:773]: shortName = phys_locations["shortName"]
 	shortname = phys_locations["shortName"]
 	please check name 
 	short_name is recomended

File: traffic_control/clients/python/trafficops/tosession.py [hunk 2]
```diff
@@ -1395,8 +1395,8 @@ def get_physical_locations(self, query_params=None):
 		:raises: Union[LoginError, OperationError]
 		"""
 
-	@api_request('post', 'phys_locations', ('3.0', '3.1', '4.0', '4.1', '5.0'))
-	def create_physical_locations(self, query_params=None, data=None):
+	@api_request('post', 'phys_locations', ('3.0', '4.0', '4.1', '5.0'))
```
Comment (ericholguin) [inline traffic_control/clients/python/trafficops/tosession.py:1398]: Please re-add 3.1

File: traffic_ops/testing/api_contract/v4/conftest.py [hunk 3]
```diff
@@ -363,6 +366,94 @@ def response_template_data(pytestconfig: pytest.Config
...
+	:returns: Verified response data
+	"""
+	if request_type == "get":
+		try:
+			api_response = api_response[0]
+			if not isinstance(api_response, list):
+				raise TypeError("malformed API response; 'response' property not an array")
+		except KeyError as e:
+			raise TypeError(f"missing API property '{e.args[0]}'") from e
+	if api_response:
```
Comment (TerekhovaKate) [inline traffic_ops/testing/api_contract/v4/conftest.py:384]: Validate api_data before returning: Currently, the code assumes that if api_response is not empty, the first element of api_response is the desired data. However, it doesn’t verify if it’s a valid data object. Consider adding validation for api_data before returning it.

**Review by TerekhovaKate — CHANGES_REQUESTED** [review]

**Review by TerekhovaKate — APPROVED** [review]
Nice work !

**Pull request merged**